docs/new-architecture-backward-turbomodule.md

```diff
@@ -0,0 +1,699 @@
+        await AuthorizationManager.getCurrentStatus();
+      setAuthorization(newState);
+    }
+    setInitialState();
+  }, []);
+
+  return (
+    <SafeAreaView>
+      <StatusBar barStyle="dark-content" />
+
+      <Button
+        title="Request Authorization"
+        onPress={() => {
+          AuthorizationManager.askAuthorization();
+        }}
+      />
+
+      <Text style={{ marginLeft: 20 }}>
+        Current authorization is: {authorization}
+      </Text>
+    </SafeAreaView>
+  );
+};
+
+export default App;
+```
+
+We can now go back to the terminal and start Metro:
+
+```sh
+npx react-native start
+```
+
+and then run the app:
+
+```sh
+npx react-native run-ios
+```
+
+After the app launches, we should see the following state in the simulator:
+
+<figure>
+  <img src="docs/assets/Turbomodules/OldArchApp.png" />
+  <figcaption>The simulator running the app</figcaption>
+</figure>
+
+But, if we tap on the button, nothing happens. That's because the `Core Location` system requires some keys in the plist. Let's do the following:
+
+1. Open the `ios/OldArchitectureApp.xcworkspace`.
+1. Select the project in the project navigator.
+1. Select the info tab.
+1. Scroll down and add the following two keys:
+   - `NSLocationWhenInUseUsageDescription`
+   - `NSLocationAlwaysAndWhenInUseUsageDescription`
+1. Add any description you want for each of these.
+
+Go back to the terminal and run `npx react-native run-ios` again. At this point, when tapping on the button, you should see this image:
+
+<figure>
+  <img src="docs/assets/Turbomodules/OldArchAppAlert.png" />
+  <figcaption>The simulator using the new module</figcaption>
+</figure>
+
+Congratulation! The Native Module is now correctly connected to your app.
+If you now relaunch the app, you should see the status getting updated.
+
+:::warning
+This is an example. Don't use this module for your location-based features.
+:::
+
+## Create the Turbomodule
+
+Let's now see how to create a Turbomodule. A Turbomodule is defined by a set of JS spec that needs to be typed. For that reason, we are going to use `Flow` dialect. `TypeScript` support is still in beta but we hope to release it soon.
+
+Once we have the JS spec, we can proceed generating the C++ code that bridges the native code with the JS counterpart.
+
+For the final step, we will implement the missing bits in the native code and then we will connect it to JS.
+
+But first, let's create a testing environment suitable for the new architecture.
+
+### Create the New Architecture Testing Environment
+
+The steps to create a New Architecture testing environment are pretty much the same as [here](#create-a-testing-environment).
+
+There are only two differences. The first one is how we create the app. We will need to run:
+
+```sh
+npx react-native init NewArchitectureApp --version next
+```
+
+The `--version next` bit will let us retrieve a version of React Native that is compatible with the new architecture. We can then proceed with:
```

Comment:
That's risky imho. This guide will then change behavior based on when you execute it. 

Ideally we should mention that version should be at least 0.68 or so.

Reply:
I tried to fix this. I kept `--version next` in the command because I couldn't find how to explicitly use the `0.68` release candidate in it, but I added some explanation in the line below. Let me know if it works! ;)